internal/controller/datadogagent/feature/orchestratorexplorer/feature.go

```diff
@@ -181,6 +185,11 @@ func (f *orchestratorExplorerFeature) ManageDependencies(managers feature.Resour
 // ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
 // It should do nothing if the feature doesn't need to configure it.
 func (f *orchestratorExplorerFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	managers.EnvVar().AddEnvVar(f.getEnabledEnvVar())
+	if !f.enabled {
+		return nil
+	}
```

Comment:
```suggestion
        // We need to explicitly add the env var to disable this feature, otherwise, we rely on Agent default
        // where the feature is enabled by default, and would error out.
	managers.EnvVar().AddEnvVar(f.getEnabledEnvVar())
	if !f.enabled {
		return nil
	}
```
Wording subject to change, but I think it's worth pointing out why this needs to be the first line, as my first thought would have been to exit early if f is not enabled